lib/matplotlib/tests/test_legend.py

```diff
@@ -508,6 +508,34 @@ def test_shadow_framealpha():
     assert leg.get_frame().get_alpha() == 1
 
 
+@image_comparison(['shadow_color.svg'])
+def test_shadow_color():
+    # Test whether a shadow of the appropriate color
+    # is generated when shadow's value is colorlike
+    fig, ax = plt.subplots()
+    ax.plot(range(100), label="test")
+    legends = {
+        'true': plt.legend(loc="upper left", shadow=True),
+        'false': plt.legend(loc="center left", shadow=False),
+        'default': plt.legend(loc="lower left"),
+        'colorstring': plt.legend(loc="upper right", shadow='red'),
+        'colortuple': plt.legend(loc="center right", shadow=(0.1, 0.2, 0.5)),
+        'colortab': plt.legend(loc="lower right", shadow='tab:cyan')
+    }
+    for x in legends:
+        plt.gca().add_artist(legends[x])
+
+
+def test_shadow_bad_type():
+    # Ensure an error is thrown if the value of shadow
+    # is neither colorlike nor bool
+    fig, ax = plt.subplots()
+    ax.plot(range(100), label="test")
+    with pytest.raises(ValueError, match="color or bool"):
+        x = plt.legend(loc="upper left", shadow="aardvark")
+        fig.canvas.draw()
```

Comment:
There's no need for this line, as the `legend` call would have raised.

Reply:
Don't think that's the case, the tests all fail.

Reply:
Oh, I didn't realize your checks are in `draw`; that's the wrong place. The setting should be validated in `__init__` so that raising occurs when the legend is created, not at some random time in the future.